Source/backend/models/Cart.js

```diff
@@ -0,0 +1,21 @@
+import mongoose from 'mongoose';
+
+const cartSchema = new mongoose.Schema({
+  products: [{
+    productId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Product',
+      required: true,
+    },
+    quantity: {
+      type: Number,
```

Comment:
I would suggest to add validation to make sure the quantity is a positive number. This will prevent wrong data from being saved.